app/scripts/filters/util.js

```diff
@@ -528,6 +528,36 @@ angular.module('openshiftConsole')
       }
       var uri = new URI(url);
       var protocol = uri.protocol();
-      return uri.is('absolute') && (protocol === 'http' || protocol === 'https');      
+      return uri.is('absolute') && (protocol === 'http' || protocol === 'https');
+    };
+  })
+  // currently used to set an alt text value for an env var that has a
+  // valueFrom attribute instead of a value.
+  .filter('altTextForValueFrom', function() {
+    return function(envVar) {
+      if(envVar.value) {
+        return;
+      }
+      if(!envVar.valueFrom) {
+        return;
+      }
+      envVar.valueIcon = 'pficon pficon-help';
+      envVar.valueIconTooltip = 'This is a referenced value that will be generated when a container is created.  On running pods you can check the resolved values by going to the Terminal tab and echoing the environment variable.';
+      _.each(envVar.valueFrom, function(from, refType) {
+        switch(refType) {
+          case 'configMapKeyRef':
+            envVar.valueAlt = 'Set to the key ' + from.key + ' in config map ' + from.name;
```

Comment:
Oh, because these are indended for a user to manipulate.  I think `$` & `$$` are sort of book keeping prefixes.  IE, I intend a user to be able to set the `valueAlt` or the `valueIcon`, so having them set `$$valueAlt` or `$$valueIcon` seems dirty...